arbnode/node.go

```diff
@@ -559,6 +561,11 @@ func createNodeImpl(
 	var stakerObj *staker.Staker
 	var messagePruner *MessagePruner
 
+	parentChainID, err := l1client.ChainID(ctx)
```

Comment:
I'm a bit worried about trusting the L1 client here. Could we pass in `nodeConfig.ParentChain.ID` from the callsite of createNode in nitro.go? I know there's other callsites, but I think it's important for the L1 RPC not to be able to trick us into signing something for the wrong chain ID.